tests/test_cloudformation/test_stack_parsing.py

```diff
@@ -181,6 +183,7 @@ def test_parse_stack_resources():
         parameters={},
         region_name="us-west-1",
     )
+    stack.initialize_resources()
```

Comment:
@dominikschubert This is probably required because we patched the CF deployment mechanism in the past, to be compatible with LocalStack.

Now that we're using our own CF deployment mechanism in LS, I'd suggest that we revert the original changes and stay compatible with moto upstream here.

In particular, this change can be undone (may require a few more changes as well):
https://github.com/localstack/moto/commit/3f32a932f846b6d055f727865d3ce0e7dc74f079

If we revert that change, we can remove the `stack.initialize_resources()` lines from the tests in this file.